core/src/main/java/com/sequenceiq/cloudbreak/core/flow2/cluster/datalake/upgrade/ClusterUpgradeActions.java

```diff
@@ -215,9 +215,11 @@ protected ClusterUpgradeContext createFlowContext(FlowParameters flowParameters,
 
             @Override
             protected void doExecute(ClusterUpgradeContext context, ClusterUpgradeFailedEvent payload, Map<Object, Object> variables) {
+                LOGGER.error("Cluster upgrade failed: {}", payload);
                 Set<Image> candidateImages = new HashSet<>();
                 Optional.ofNullable(getCurrentImage(variables)).ifPresent(si -> candidateImages.add(si.getImage()));
                 Optional.ofNullable(getTargetImage(variables)).ifPresent(si -> candidateImages.add(si.getImage()));
+                clusterUpgradeService.handleUpgradeClusterFailure(payload.getResourceId(), payload.getException().getMessage(), payload.getDetailedStatus());
```

Comment:
Would be worth covering this with an UT.

Reply:
Currently, there is no unit test for this class so I'll create a followup jira to cover the whole class with unit tests.